### PR TITLE
Initialize project

### PR DESCRIPTION
**Pull request by AndrewGable**
### Details
First PR for the new GitHub-Actions repo that will house all shared action code

### Related Issues
Related https://github.com/Expensify/Expensify/issues/432173

### Manual Tests
Going to test on the `Expensify/eslint-config-expensify` repo.

### CODE REVIEW

File: .github/workflows/CLA.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          PERSONAL_ACCESS_TOKEN : ${{ secrets.CLA_BOTIFY_TOKEN }}
+        with:
+          path-to-signatures: '${{ github.repository }}/cla.json'
+          path-to-document: 'https://github.com/${{ github.repository }}/blob/main/contributingGuides/CLA.md'
+          branch: 'main'
+          remote-organization-name: 'Expensify'
+          remote-repository-name: 'CLA'
+          lock-pullrequest-aftermerge: false
+          allowlist: OSBotify,snyk-bot
```
Comment (johnmlee101) [inline .github/workflows/CLA.yml]: newline
```suggestion
          allowlist: OSBotify,snyk-bot

```

File: .github/scripts/shellCheck.sh [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+  fi
+done
+
+cd "$CURRENT_DIR" || exit 1
+
+if [ $EXIT_CODE == 0 ]; then
+  success "ShellCheck passed for all files!"
+fi
+
+exit $EXIT_CODE
```
Comment (johnmlee101) [inline .github/scripts/shellCheck.sh]:
```suggestion
exit $EXIT_CODE

```

File: .github/workflows/npmPublish.yml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      - name: Publish to npm
+        run: npm publish
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+
+      - name: Comment on PR
+        run: |
+          gh pr comment ${{ inputs.pull_request_number }} --repo ${{ inputs.repository }} --body ":rocket: Published in ${{ env.NEW_VERSION }}"
+        env:
+          GITHUB_TOKEN: ${{ github.token }}
```
Comment (johnmlee101) [inline .github/workflows/npmPublish.yml]:
```suggestion
          GITHUB_TOKEN: ${{ github.token }}

```

File: .github/workflows/shellCheck.yml [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+jobs:
+  lint:
+    if: ${{ github.actor != 'OSBotify' || github.event_name == 'workflow_call' }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Lint shell scripts with ShellCheck
+        run: npm run shellCheck
```
Comment (johnmlee101) [inline .github/workflows/shellCheck.yml]:
```suggestion
        run: npm run shellCheck

```

File: .github/scripts/shellCheck.sh [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+    # Special case to remove any starting '//' when the input path was absolute
+    abs_path=${abs_path/#\/\//\/}
+
+    echo "$abs_path"
+}
+
+declare -r DIRECTORIES_TO_IGNORE=(
+  './node_modules'
+  './vendor'
+  './ios/Pods'
```
Comment (johnmlee101) [inline .github/scripts/shellCheck.sh:115]: do we have to maintain this list on _all_ repos we apply this to?

**Review by johnmlee101 — CHANGES_REQUESTED** [review]

File: .github/scripts/shellCheck.sh (hunk 5 above)
Comment (AndrewGable) [reply .github/scripts/shellCheck.sh:115]: No - this is only shell checking this repo in particular. This is also a bit of a pre-optimization as we don't have any shell files yet.

**Review by johnmlee101 — APPROVED** [review]

**Pull request merged**